src/builtins.zig

```diff
@@ -179,9 +181,25 @@ test "fs" {
     try ndir.setAsCwd();
 }
 
-/// TODO implement real version of exit
-fn exit(h: *HSH, i: *ParsedIterator) Err!u8 {
-    return noimpl(h, i);
+fn exit(hsh: *HSH, i: *ParsedIterator) Err!u8 {
+    std.debug.assert(std.mem.eql(u8, "exit", i.first().cannon()));
+    var code: usize = 0;
+    if (i.next()) |next| {
+        const parsed_code = std.fmt.parseInt(isize, next.cannon(), 10) catch |err| {
+            log.err("Failed to parse exit code because {}\n", .{err});
+            return err;
+        };
+        code = @bitCast(parsed_code);
```

Comment:
is intcast, or bitcast better here? I'm actually asking, I haven't memorized the semantics of intcast yet, bitcast might be better?

Reply:
Bitcast preserves the bits when converting between integer types, which was my intention here. I wanted to mimic the wraparound behavior that we get in C. I read in the value as an isize so the user can input a large negative number if they want to. Then, bitcast will convert bits into an unsigned value and I use mod 256 to constrain the value between 0 and 255, which I can truncate into a u8 without losing any data.

It's a bit of a hacky way to mimic how C would allow the exit code to under/overflow. Let me know if you think of something more elegant.

Reply:
Maybe we can trust parseInt to handle negatives and large values for us? That might be an alternative.

Reply:
no, I think I agree with you, we're casting to u8 so we want' exclusively bitcast behavior here.